### PR TITLE
[Merged by Bors] - chore: Define the class `IsZlattice`

### PR DESCRIPTION
**Pull request by xroblot**
See the Zulip [thread](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Class.20.60IsZlattice.60/near/426301607)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by xroblot** [comment]
!bench

File: Mathlib/Algebra/Module/Zlattice.lean [hunk 1]
```diff
@@ -409,23 +426,31 @@ theorem Zlattice.FG : AddSubgroup.FG L := by
     exact fg_span (LinearIndependent.setFinite h_lind)
 
 theorem Zlattice.module_finite : Module.Finite ℤ L :=
-  Module.Finite.iff_addGroup_fg.mpr ((AddGroup.fg_iff_addSubgroup_fg L).mpr (FG K hs))
+  Module.Finite.iff_addGroup_fg.mpr ((AddGroup.fg_iff_addSubgroup_fg L).mpr (FG K L))
+
+instance instModuleFinite_Zlattice {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
+    [FiniteDimensional ℝ E] [ProperSpace E] (L : AddSubgroup E) [DiscreteTopology L]
+    [IsZlattice ℝ L] : Module.Finite ℤ L := Zlattice.module_finite ℝ L
```
Comment (sgouezel) [inline Mathlib/Algebra/Module/Zlattice.lean:433]: Do you really need L to be a lattice for this instance ? Being discrete should be enough, I guess.

File: Mathlib/Algebra/Module/Zlattice.lean [hunk 2]
```diff
@@ -409,23 +426,31 @@ theorem Zlattice.FG : AddSubgroup.FG L := by
...
 
 theorem Zlattice.module_free : Module.Free ℤ L := by
```
Comment (sgouezel) [inline Mathlib/Algebra/Module/Zlattice.lean:435]: Here again, this is probably true without the lattice instance, being discrete should be enough.

Comment (xroblot) [reply Mathlib/Algebra/Module/Zlattice.lean:435]: Yes, that's a very good point. The class `IsZlattice` is for full lattices but these results are true for lattices in general. I'll fix that. Thanks!

Comment (xroblot) [reply Mathlib/Algebra/Module/Zlattice.lean]: Ok. I pushed a new version without the `IsZlattice` hypothesis.

**Comment by riccardobrasca** [comment]
Can you add a TODO explaining that sooner or later we want a version not over `Z`? Thanks!

bors d+

**Pull request closed without merging**